File: src/main/java/in/costea/wiles/exceptions/UnexpectedTokenException.java
package in.costea.wiles.exceptions;

import in.costea.wiles.data.TokenLocation;
import org.jetbrains.annotations.NotNull;

public class UnexpectedTokenException extends AbstractCompilationException
{
    public UnexpectedTokenException(@NotNull String s, TokenLocation where)
    {
        super("Unexpected token: " + s, where);
    }
}

File: src/main/java/in/costea/wiles/exceptions/UnexpectedTokenException.kt
package `in`.costea.wiles.exceptions

import `in`.costea.wiles.data.TokenLocation
import `in`.costea.wiles.constants.ErrorMessages.UNEXPECTED_TOKEN_ERROR

class UnexpectedTokenException(s: String, where: TokenLocation?) :
    AbstractCompilationException(UNEXPECTED_TOKEN_ERROR.format(s), where)